heron/tools/cli/src/python/response.py

```diff
@@ -68,33 +70,55 @@ def add_context(self, err_context, succ_context=None):
     self.err_context = err_context
     self.succ_context = succ_context
 
+def do_log(log, msg):
+  if msg:
+    log(msg)
+
+def log_context(resp):
+  status = resp.status
+  if status == Status.Ok or status == Status.DryRun:
+    do_log(Log.info, resp.succ_context)
+  elif status == Status.HeronError:
+    do_log(Log.error, resp.err_context)
+  elif status == Status.InvocationError:
+    pass
+  else:
+    raise RuntimeError("Unknown status type of value %d", status)
+
+# # pylint: disable=too-many-branches,unreachable
 def render(resp):
-  def do_log(log, msg):
-    if msg:
-      log(msg)
+  """ Render output of the process until it exits
+
+  :param resp: Response
+  :return:
+  """
   if isinstance(resp, list):
```

Comment:
This `isinstance` of list or Response seems like some python jank. Are there places where lists of Response objects are being passed in? If not, we should remove this.